fix(angular-loader): do not depend on "closure" globals that may not be available

gkalpak opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix.


**What is the current behavior? (You can also link to an open issue here)**
See #15880.


**What is the new behavior (if this is a feature change)?**
Code that is distributed as part of both `angular.js` and `angular-loader.js` does not depend on "closure" globals that may not be available in `angular-loader`.


**Does this PR introduce a breaking change?**
No.


**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:
Fixes #15880.

gkalpak committed:
fix(angular-loader): do not depend on "closure" globals that may not be available

Code that is distributed as part of both `angular.js` and `angular-loader.js`
should not depend on "closure" globals that may not be available in
`angular-loader`.

Fixes #15880

Narretz commented on src/stringify.js:
```diff
@@ -9,7 +13,10 @@ function serializeObject(obj, maxDepth) {
   // based on this specific depth and then stringify it.
   if (isValidObjectMaxDepth(maxDepth)) {
-    obj = copy(obj, null, maxDepth);
+    if (!copyFn) {
```
Shouldn't `angular.copy` be always defined here?

Narretz commented on src/loader.prefix:
```diff
@@ -7,5 +7,24 @@
+    function isScope(obj) {return obj && obj.$evalAsync && obj.$watch;}
+    function isWindow(obj) {return obj && obj.window === obj;}
+    function sliceArgs(args, startIndex) {return Array.prototype.slice.call(args, startIndex || 0);}
+    function toJsonReplacer(key, value) {
```
Should we add a note where this and other functions are used? This makes it easier later to check if they are still necessary after refactorings etc.

gkalpak replied:
You mean add a comment here? Or in the places where these are used?

Narretz replied:
In this file - a note at the top is probably enough, saying something of the effect that these functions are defined here because they must be available in angular-loader without angular.js loaded, and that they are copied from the Angular.js file

Narretz reviewed (approved):
A few comments, otherwise LGTM

gkalpak committed:
fixup! fix(angular-loader): do not depend on "closure" globals that may not be available

gkalpak commented:
Rebased, addressed comments and [fixed a bug][1]. Will merge as soon as Travis is happy too.

[1]: https://github.com/angular/angular.js/pull/15881/commits/c7e48c05b3b6178e61a95ababc1dc2d13c2eeef4#diff-c2636daee36d92d1958c5fcc6cbd0554R17

Narretz commented:
LGTM. Looks like nobody uses the loader ...
